src/JSON.jl

```diff
@@ -138,7 +138,7 @@ end
 
 function _print(io::IO, state::State, a)
     start_object(io, state, true)
-    range = typeof(a).names
+    range = @compat fieldnames(a)
```

Comment:
You don't need `@compat` for function calls, you only need it for rewriting syntax.  Just `fieldnames(a)` should be sufficient to call the Compat version in older Julia releasese.